### ISSUE TITLE
Http push AWS SNS integration invalid signature on spaces in external message values

### ISSUE THREAD

**Issue by alstanchev** [eclipse-ditto/ditto#1705]
In the case of using [AWS SNS integration](https://eclipse.dev/ditto/connectivity-hmac-signing.html#aws-sns) and trying to send all data there is the possibility that in the headers there could be `accept-language:en, en-GB;q=0.9, bg;q=0.8, en-US;q=D0.7`.
The spaces however lead to improper handling of the query on AWS side. When decoded there the signature calculated by AWS doesn't match the signature calculated on ditto side [AwsRequestSigning.java](https://github.com/eclipse-ditto/ditto/blob/58ce86b14aef9c43af2d153d0e4d5fa134065911/connectivity/service/src/main/java/org/eclipse/ditto/connectivity/service/messaging/httppush/AwsRequestSigning.java#L93C2-L93C99)

Error response:
`Ran into a failure when publishing signal: Got non success status code: <403> and body: <\"<ErrorResponse xmlns=\\\"http://sns.amazonaws.com/doc/2010-03-31/\\\">\\n  <Error>\\n    <Type>Sender</Type>\\n    <Code>SignatureDoesNotMatch</Code>\\n    <Message>The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details.</Message>\\n  </Error>\\n</ErrorResponse>\\n\">`

**Comment by thjaeckle** [comment]
> One approach could be to encode "=" char in header values before it is sent to the payload mappers [here](https://github.com/eclipse-ditto/ditto/blob/58ce86b14aef9c43af2d153d0e4d5fa134065911/connectivity/service/src/main/java/org/eclipse/ditto/connectivity/service/messaging/OutboundMappingProcessor.java#L166)

That seems like the wrong place IMO. 
Just because we have for AWS SNS special escaping requirements we should not adjust basically all payload mappers.  
Because reversing the escaping would have to be done by everyone receiving events by Ditto. 
This would even be a breaking change for connections and protocols where this is not an issue and `=` is already used in headers. 

> or another fix could be to update the example of the script in  [AWS SNS integration](https://eclipse.dev/ditto/connectivity-hmac-signing.html#aws-sns) to encode the dittoHeaders like:

Sure, that would be the easiest solution. 
But still decoding would be required on the other side of SNS. 

Or even not to map all incoming headers as Ditto headers to send to SNS, only selected ones.  
Thinking about that, this is rather dumb to just send all headers without sanatizing them.

**Comment by alstanchev** [comment]
Makes sense, by sanitising do you mean in the payload mapper or something else?

**Comment by thjaeckle** [comment]
Yes, in the payload mapper, only selectively add headers.

So instead of:
```javascript
JSON.stringify(
  Ditto.buildDittoProtocolMsg(
    namespace, 
    name, 
    group, 
    channel, 
    criterion, 
    action, 
    path, 
    dittoHeaders, 
    value, 
    status, 
    extra
  )
);
```

```javascript
let headers = {
  "correlation-id" = dittoHeaders["correlation-id"],
  foo = dittoHeaders.foo,
  ...
};
JSON.stringify(
  Ditto.buildDittoProtocolMsg(
    namespace, 
    name, 
    group, 
    channel, 
    criterion, 
    action, 
    path, 
    headers, 
    value, 
    status, 
    extra
  )
);
```
